aptos-move/framework/aptos-framework/sources/datastructures/big_ordered_map.spec.move

```diff
@@ -0,0 +1,250 @@
+        pragma opaque;
+        pragma verify = false;
+        aborts_if [abstract] forall i in 0..len(keys), j in 0..len(keys) where i != j : keys[i] == keys[j];
+        ensures [abstract] forall i in 0..len(keys), j in 0..len(keys) where i != j : keys[i] != keys[j];
+        ensures [abstract] forall k: K {spec_contains_key(result, k)} : vector::spec_contains(keys,k) <==> spec_contains_key(result, k);
+        ensures [abstract] forall i in 0..len(keys) : spec_get(result, keys[i]) == values[i];
+        ensures [abstract] spec_len(result) == len(keys);
+    }
+
+    spec upsert {
+        pragma opaque;
+        pragma verify = false;
+        ensures [abstract] !spec_contains_key(old(self), key) ==> option::is_none(result);
+        ensures [abstract] spec_contains_key(self, key);
+        ensures [abstract] spec_get(self, key) == value;
+        ensures [abstract] spec_contains_key(old(self), key) ==> ((option::is_some(result)) && (option::spec_borrow(result) == spec_get(old(
+            self), key)));
+        ensures [abstract] !spec_contains_key(old(self), key) ==> spec_len(old(self)) + 1 == spec_len(self);
+        ensures [abstract] spec_contains_key(old(self), key) ==> spec_len(old(self)) == spec_len(self);
+        ensures [abstract] forall k: K: spec_contains_key(old(self), k) && k != key ==> spec_get(old(self), k) == spec_get(self, k);
+        ensures [abstract] forall k: K: spec_contains_key(old(self), k) ==> spec_contains_key(self, k);
+    }
+
+    spec add_all {
+        pragma opaque;
+        pragma verify = false;
+    }
+
+    spec borrow_front<K: drop + copy + store, V: store>(self: &BigOrderedMap<K, V>): (K, &V) {
+        pragma opaque;
+        pragma verify = false;
+        ensures [abstract] spec_contains_key(self, result_1);
+        ensures [abstract] spec_get(self, result_1) == result_2;
+        ensures [abstract] forall k: K where k != result_1: spec_contains_key(self, k) ==>
+        std::cmp::compare(result_1, k) == std::cmp::Ordering::Less;
+    }
+
+    spec borrow_back {
+        pragma opaque;
+        pragma verify = false;
+        ensures [abstract] spec_contains_key(self, result_1);
+        ensures [abstract] spec_get(self, result_1) == result_2;
+        ensures [abstract] forall k: K where k != result_1: spec_contains_key(self, k) ==>
+        std::cmp::compare(result_1, k) == std::cmp::Ordering::Greater;
+    }
+
+    spec pop_front<K: drop + copy + store, V: store>(self: &mut BigOrderedMap<K, V>): (K, V) {
+        pragma opaque;
+        pragma verify = false;
+    }
+
+    spec pop_back {
+        pragma opaque;
+        pragma verify = false;
+    }
+
+    spec prev_key<K: drop + copy + store, V: store>(self: &BigOrderedMap<K, V>, key: &K): Option<K> {
+        pragma opaque;
+        pragma verify = false;
+        ensures [abstract] result == std::option::spec_none() <==> (spec_contains_key(self, key) &&
```

Comment:
The `none` case here can be simplified:

`        ensures [abstract] result == std::option::spec_none() <==> (forall k: K {spec_contains_key(self, k)} where spec_contains_key(self, k) && k != key:
              std::cmp::compare(key, k) == std::cmp::Ordering::Less);`

Reply:
done